src/go/rpk/pkg/redpanda/version.go

```diff
@@ -22,7 +22,7 @@ func VersionFromString(s string) (Version, error) {
 	//   - index 1: the Year
 	//   - index 2: the Feature
 	//   - index 3: the Patch
-	vMatch := regexp.MustCompile(`^v?(\d{2})\.(\d{1,2})\.(\d{1,2})(?:\s|$)`).FindStringSubmatch(s)
+	vMatch := regexp.MustCompile(`^v?(\d{2})\.(\d{1,2})\.(\d{1,2})(?:\s|-rc\d{1,2}|$)`).FindStringSubmatch(s)
```

Comment:
we'll probably only ever have up to 99 release candidates, but this is a bit prescriptive to the format here.